### PR TITLE
RWA-87-Get staff location

### PR DESCRIPTION
**Pull request by lewis150193**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-87

### Change description ###
Get staff location based on hearing centre

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by iguannago — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/utils/GetStaffLocation.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            case "manchester":
+                return "Manchester";
+            case "newcastle":
+                return "Newcastle";
+            case "newport":
+                return "Newport";
+            case "taylorHouse":
+                return "Taylor House";
+            default:
+                return null;
```
Comment (alexbaeza) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/utils/GetStaffLocation.java:35]: Maybe this should throw an exception

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/utils/GetStaffLocation.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.iacaseapi.infrastructure.utils;
+
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.HearingCentre;
+
+public class GetStaffLocation {
```
Comment (alexbaeza) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/utils/GetStaffLocation.java:5]: We should probably rename this class so it's something like StaffLocation.getLocation()

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DeriveHearingCentreHandlerTest.java [hunk 3]
```diff
@@ -110,13 +111,16 @@ public void should_not_set_hearing_centre_if_already_exists() {
         when(callback.getEvent()).thenReturn(Event.SUBMIT_APPEAL);
         when(caseDetails.getCaseData()).thenReturn(asylumCase);
         when(asylumCase.read(HEARING_CENTRE)).thenReturn(Optional.of(existingHearingCentre));
+        when(asylumCase.read(HEARING_CENTRE)).thenReturn(Optional.of(GetStaffLocation.getStaffLocation(existingHearingCentre)));
```
Comment (alexbaeza) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DeriveHearingCentreHandlerTest.java:114]: should this be STAFF_LOCATION?

**Review by alexbaeza — CHANGES_REQUESTED** [review]

**Review by iguannago — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/utils/GetStaffLocation.java (hunk 2 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/utils/GetStaffLocation.java]: agree

**Review by alexbaeza — APPROVED** [review]

**Review by BenNzewi — CHANGES_REQUESTED** [review]
Coventry Magistrates Court, Nottingham Justice Centre & Glasgow Tribunals Centre are missing tests

**Review by BenNzewi — APPROVED** [review]
Coventry Magistrates Court, Nottingham Justice Centre & Glasgow Tribunals Centre tests + functional tests added

**Pull request merged**